fnet/src/vespa/fnet/frt/require_capabilities.cpp

```diff
@@ -15,15 +16,22 @@ FRT_RequireCapabilities::allow(FRT_RPCRequest& req) const noexcept
 {
     const auto& auth_ctx = req.GetConnection()->auth_context();
     const bool is_authorized = auth_ctx.capabilities().contains_all(_required_capabilities);
-    if (!is_authorized) {
+    if (is_authorized) {
+        return true;
+    } else {
+        const auto mode = capability_enforcement_mode_from_env();
+        if (mode == CapabilityEnforcementMode::Disable) {
+            return true;
+        }
         auto peer_spec = req.GetConnection()->GetPeerSpec();
         std::string method_name(req.GetMethodName(), req.GetMethodNameLen());
-        LOGBT(warning, peer_spec, "Permission denied for RPC method '%s'. "
+        LOGBT(warning, peer_spec, "%sPermission denied for RPC method '%s'. "
                                   "Peer at %s with %s. Call requires %s, but peer has %s",
+              ((mode == CapabilityEnforcementMode::LogOnly) ? "(Dry-run only, not enforced): " : ""),
               method_name.c_str(), peer_spec.c_str(),
               to_string(auth_ctx.peer_credentials()).c_str(),
               _required_capabilities.to_string().c_str(),
               auth_ctx.capabilities().to_string().c_str());
+        return (mode == CapabilityEnforcementMode::Enforce) ? false : true;
```

Comment:
prefer logic expression over conditional mapping of bool to bool

Reply:
Fixed